### ISSUE TITLE
Add ability to generate superpixels and label each superpixel

### ISSUE THREAD

**Issue by rsnk96** [opencv/cvat#772]
It would be nice to have a feature to generate superpixels, and label superpixels in one mouse-click (sort of like a brush, but it labels the superpixel if the mouse has hovered over with the Mouse Button pressed down). An easy idea for this is to run K-means in RGB+(x,y) space, like the SLIC/[SLIC-zero](https://ivrl.epfl.ch/research-2/research-current/research-superpixels/#SLICO) algorithm does

A sample implementation of SLIC in C++ is available [here](https://github.com/PSMM/SLIC-Superpixels), this can be ported to OpenCV.js

**Comment by nmanovic** [comment]
@rsnk96 , did you try to annotate data using the approach? What is the annotation speed in the case? I like new features but I need some information how to prioritize them. Does the approach give better quality or better speed than traditional methods?

**Comment by rsnk96** [comment]
This is a feature that is available in [LabelBox](https://medium.com/labelbox/introducing-best-in-class-image-annotation-tools-for-computer-vision-applications-de5da8931d6e), sample usage:
![superpixel-annotation-in-labelbox](https://miro.medium.com/max/600/1*zafieoh5P4Q2fI-qOv9gEA.gif)

I believe this would give better speed in the following cases:
1. Where the objects do not have clear boundaries (hence creating problems in DEXTR/grabcut(not implemented yet)/traditional approaches )
2. When the object has a very non-smooth error (hence drawing a polygon would be painful)
3. In cases where an object has enclaves in between which belong to a different class/no-class

**Comment by jrreyes29** [comment]
> here

How do you can port this to opencvjs because opencvjs don't support this method createSuperpixelLSC()

**Comment by rsnk96** [comment]
I'm not quite sure I followed you @jrreyes29. Where is that method being called?

**Comment by jrreyes29** [comment]
Hi @rsnk96 the method in opencv is this one https://docs.opencv.org/4.1.2/df/d6c/group__ximgproc__superpixel.html#gga0024d8bac96a9f0d49bd97665d76ae08a9b397d2eb2d75580b3a481bf8e0edef1

**Comment by rsnk96** [comment]
Ah, didn't know that it comes as a part of `ximgproc` now. 

I guess that leaves us with no option but to implement the superpixel method in JS from scratch...

**Comment by jrreyes29** [comment]
but implement that in js from scratch take long time hahahaha that is the idea about opencvjs

**Comment by luiztauffer** [comment]
superpixel annotation would be awesome!

**Comment by patzhang** [comment]
[Here](https://github.com/kyamagu/js-segment-annotator) is an implementation in javascript for Superpixels using SLIC. It has a nice [demo](http://kyamagu.github.io/js-segment-annotator/) that you can try online.

@nmanovic  Is the demo good enough to compare?

Also, the [COCO-Stuff dataset](https://arxiv.org/pdf/1612.03716.pdf) used superpixels (shown in the paper) in their [annotation tool](https://github.com/nightrome/cocostuff10k#annotation-tool).

**Comment by Hommus** [comment]
@nmanovic 
Any ETA on this feature? 

How would one go about implementing a new tool / feature like this? I have created it in Python, I'm just struggling to re-implement it in javascript.

**Comment by nmanovic** [comment]
I don't think we need the feature anymore. SAM/SAMv2/SAMv3 completely covers the scenario and works much better.
https://www.cvat.ai/resources/changelog/sam-3-image-segmentation
https://docs.cvat.ai/docs/annotation/auto-annotation/ai-tools/#interactors
